docs/security-considerations.rst

```diff
@@ -181,13 +181,14 @@ Sending and Receiving Ether
      contract. Again, the best practice here is to use a :ref:`"withdraw"
      pattern instead of a "send" pattern <withdrawal_pattern>`.
 
-Callstack Depth
-===============
+Call stack Depth
+================
 
 External function calls can fail any time because they exceed the maximum
 call stack of 1024. In such situations, Solidity throws an exception.
 Malicious actors might be able to force the call stack to a high value
-before they interact with your contract.
+before they interact with your contract, by controlling the number of calls
+and local function variables on the stack.
```

Comment:
@miohtama Do you have an example on the effect of local function variables?

Reply:
I don't think local stack items would have an effect on call stack depth.

Reply:
Thanks for updating this. 

Does anyone know where the `call stack` is implemented in the source code ? I can see where each stack for each function gets created, but I can't see where it uses 1024 limit check...

Reply:
The call stack is a feature of the EVM, it is not implemented in the solidity compiler.

Reply:
Yes, I am looking at the `evm.go` or any other `go` files related to `evm`. Seems like `call stack` itself is not implemented and they only count external calls.

Reply:
Yes, this is only about external calls, that's why this paragraph starts with "External function calls".